src/calc/src/expr_manager.rs

```diff
@@ -0,0 +1,540 @@
+}
+
+impl ToExpr for Opt {
+    #[rustfmt::skip]
+    fn to_expr(&self) -> Option<ExprItem> {
+        Some(match self {
+            //                             Disp         Eval     Priority LeftAsoc SkipConv
+            Self::Add =>     ExprItem::new("+",         "+",        0,      true,   true),
+            Self::Sub =>     ExprItem::new("-",         "-",        0,      true,   true),
+            Self::Mul =>     ExprItem::new("⋅",         "*",        1,      true,   true),
+            Self::Div =>     ExprItem::new("÷",         "/",        1,      true,   true),
+            Self::Sin =>     ExprItem::new("sin ",      "sin",      3,      false,  false),
+            Self::Cos =>     ExprItem::new("cos ",      "cos",      3,      false,  false),
+            Self::Tg =>      ExprItem::new("tg ",       "tg",       3,      false,  false),
+            Self::Cotg =>    ExprItem::new("cotg ",     "cotg",     3,      false,  false),
+            Self::Arcsin =>  ExprItem::new("sin⁻¹ ",    "arcsin",   3,      false,  false),
+            Self::Arccos =>  ExprItem::new("cos⁻¹ ",    "arccos",   3,      false,  false),
+            Self::Arctg =>   ExprItem::new("tg⁻¹ ",     "arctg",    3,      false,  false),
+            Self::Arccotg => ExprItem::new("cotg⁻¹ ",   "arccotg",  3,      false,  false),
+            Self::Log =>     ExprItem::new("log ",      "log10",    3,      false,  false),
+            Self::LogN =>    ExprItem::new("logₙ ",     "log",      2,      true,   false),
+            Self::Ln =>      ExprItem::new("ln ",       "ln",       3,      false,  false),
+            Self::Sqrt =>    ExprItem::new("√",         "sqrt",     3,      false,  false),
+            Self::Root =>    ExprItem::new("ⁿ√",        "root",     2,      true,   false),
+            Self::Root3 =>   ExprItem::new("³√",        "root",     3,      false,  false),  // FIXME: How am I gonna handle this?
+            Self::Pow =>     ExprItem::new("^",         "^",        3,      true,   true),
+            Self::Pow2 =>    ExprItem::new("²",         "^2",       3,      true,   true),
+            Self::Abs =>     ExprItem::new("abs ",      "abs",      3,      false,  false),
+            Self::Comb =>    ExprItem::new("C",         "comb",     2,      true,   false),
+            Self::Fact =>    ExprItem::new("!",         "!",        3,      true,   true),
+            Self::Mod =>     ExprItem::new("mod",       "mod",      2,      true,   true),
+        })
+    }
+}
+
+impl ToExpr for PressedButton {
+    #[rustfmt::skip]
+    fn to_expr(&self) -> Option<ExprItem> {
+        Some(match self {
+            Self::Num(num) => {
+                // Convert number to digit.
+                let s = match *num >= 10 {
+                    // A-F
+                    true => char::from_digit(*num as u32, 16).unwrap().to_string().to_uppercase(),
+                    // 0-9
+                    false => num.to_string()
+                };
```

Comment:
The match statement is useless here since the `char::from_digit` also handles the number in range of <0, 9>.
Also use `char::to_ascii_uppercase` saves some cpu clocks as well.
Something like this
```rs
let s = char::from_digit(*num as u32, 16).unwrap().to_ascii_uppercase().to_string();
```

Reply:
Got it!